### DISCUSSION TITLE
0.23 JSVM cannot find Buffer.isBuffer() method (used to work on 0.22)

### DISCUSSION THREAD

**Discussion by Zhomart** [pocketbase/pocketbase#5936] [Q&A]
I just recently started using pocketbase with version 0.22, I don't have much code. But I wanted to upgrade to 0.23 for batch API.

I have a script that uses javascript "cheerio" library to parse HTML and store the content in pb. This used to work in 0.22, but fails in 0.23.  It looks like it's using "parse.ts" file (typescript), even though I'm loading the js file (see the example code below).

Here are more details:

```
$ pocketbase hello

panic: TypeError: Object has no member 'isBuffer' at parse (/Users/user/Work/finder/node_modules/cheerio/src/parse.ts:39:57(13))

goroutine 66 [running]:
github.com/dop251/goja.(*Runtime).toReflectValue.(*Runtime).wrapJSFunc.func1({0x14000c180f0?, 0x2?, 0x4?})
        /home/runner/go/pkg/mod/github.com/dop251/goja@v0.0.0-20241009100908-5f46f2705ca3/runtime.go:2287 +0x624
github.com/spf13/cobra.(*Command).execute(0x140006f0c08, {0x1050533a0, 0x0, 0x0})
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:989 +0x81c
github.com/spf13/cobra.(*Command).ExecuteC(0x140006f0f08)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1117 +0x344
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1041
github.com/pocketbase/pocketbase.(*PocketBase).Execute.func2()
        /home/runner/work/pocketbase/pocketbase/pocketbase.go:179 +0x2c
created by github.com/pocketbase/pocketbase.(*PocketBase).Execute in goroutine 1
        /home/runner/work/pocketbase/pocketbase/pocketbase.go:177 +0xe0
```
**cheerio** - version "1.0.0"

failing library code - [node_modules/cheerio/src/parse.ts:39](https://github.com/cheeriojs/cheerio/blob/main/src/parse.ts#L39)

```
  return function parse(
    content: string | Document | AnyNode | AnyNode[] | Buffer,
    options: InternalOptions,
    isDocument: boolean,
    context: ParentNode | null,
  ): Document {
    if (typeof Buffer !== 'undefined' && Buffer.isBuffer(content)) {
      content = content.toString();
    }
    ...
```

- load-parse.js - https://gist.github.com/Zhomart/69b3de2b769b278f3fad8ad2caa0827d
- 

my code:

```
function parseHtml(body) {
    const cheerioLoadParse = require(`${__hooks}/../node_modules/cheerio/dist/commonjs/load-parse.js`)
    const $ = cheerioLoadParse.load(body)
    console.log(`cheerio loaded the html`)
    console.log($("title").text().trim())
}
```

**Comment by ganigeorgiev** [comment]
v0.23.0 enables the [`Buffer` goja_nodejs](https://github.com/dop251/goja_nodejs/blob/master/buffer/buffer.go) module but its implementation seems incomplete.

It probably worked with v0.22 because `Buffer` was `undefined` and wasn't executing the failed condition.

I don't have time at the moment to contribute in `goja_nodejs` for this (I'm not sure what other methods are also missing) and if this ended up causing too much troubles to other users I may disable the `Buffer` module.

For now you can try to open an issue in the upstream library for the missing method OR overwrite the global `Buffer` with undefined to try emulating the old behavior.

**Reply by Zhomart** [reply]
Thanks @ganigeorgiev , this worked!

In: main.pb.js
```
// Fix for: https://github.com/pocketbase/pocketbase/discussions/5936
Buffer = undefined
```
